sdk-extensions/logging/src/main/java/io/opentelemetry/sdk/logging/export/BatchLogProcessor.java

```diff
@@ -209,8 +209,16 @@ public void run() {
 
     private CompletableResultCode forceFlush() {
       CompletableResultCode flushResult = new CompletableResultCode();
-      this.flushRequested.compareAndSet(null, flushResult);
-      return this.flushRequested.get();
+      if (this.flushRequested.compareAndSet(null, flushResult)) {
+        return flushResult;
+      }
+      flushResult = this.flushRequested.get();
```

Comment:
Appears to have been a real, though highly improbable, race, caught by the nullness checker

Reply:
Actually realized it's reasonable probableness - a pending flush from before this was invoked could rather easily complete between the two lines of code I guess.